mesonbuild/modules/i18n.py

```diff
@@ -12,11 +12,20 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from .. import coredata, mesonlib, build
+from .. import coredata, mesonlib, build, dependencies, mlog
 import sys
 
 class I18nModule:
 
+    def has_gettext_tools(self):
+       have_all_tools = True
+       gettext_programs = ['xgettext', 'msgfmt', 'msgmerge']
+       for prog in gettext_programs:
+           dep = dependencies.ExternalProgram(prog)
+           if not dep.found():
+               have_all_tools =False
+       return have_all_tools
```

Comment:
Forgive my ignorance, but by caching you mean making it a global, or is it something else?